kernel/src/syscall.rs

```diff
@@ -252,53 +255,53 @@ impl Syscall {
     pub fn from_register_arguments(
         syscall_number: u8,
         r0: usize,
-        r1: usize,
-        r2: usize,
-        r3: usize,
+        r1: MetaPtr,
+        r2: MetaPtr,
+        r3: MetaPtr,
     ) -> Option<Syscall> {
```

Comment:
If we can't use `usize` here, then I think we need a new type (`RegisterData`?) as registers don't necessarily hold pointers.

Reply:
I think that would, from an interface point of view, a nicer spelling.

I just wonder if we will end up with too many wrapping types. Possibly just the one more, a `RegisterData` as you suggest which is the correct types for this function and for the saved register file.

Reply:
I'm not currently thinking of `MetaPtr` as a wrapping type, but as a type with a very specific meaning within Tock (as https://github.com/tock/book/pull/53 will hopefully eventually articulate).

Something like `RegisterData` would be a wrapper for whatever is register sized. Or it could be `ureg`? `uregister`? `umachine`?

Reply:
I have added a doc comment to MetaPtr that it is at least as wide as `usize` and `*const ()` so can also be used for arguments that are unions of these.

I can add a `RegisterData`, but I think this might be slightly wrong. Some CHERI machines have had distinct registers to contain `usize` and `MetaPtr`, for instance. What is really happening here is that we have possibly padded usize/u32/*const() to make MetaPtr in userspace (as it is the widest), and this decoding is removing padding as appropriate.